core/savings.go

```diff
@@ -18,32 +20,90 @@ type publisher interface {
+		fmt.Printf("unable to save %s settings.\n", key)
+	}
+}
+
 // Site is the main configuration container. A site can host multiple loadpoints.
 type Savings struct {
 	clock                          clock.Clock
 	tariffs                        tariff.Tariffs
-	started                        time.Time // Boot time
-	updated                        time.Time // Time of last charged value update
-	gridCharged                    float64   // Grid energy charged since startup (kWh)
-	gridCost                       float64   // Running total of charged grid energy cost (e.g. EUR)
-	gridSavedCost                  float64   // Running total of saved cost from self consumption (e.g. EUR)
-	selfConsumptionCharged         float64   // Self-produced energy charged since startup (kWh)
-	selfConsumptionCost            float64   // Running total of charged self-produced energy cost (e.g. EUR)
-	lastGridPrice, lastFeedInPrice float64   // Stores the last published grid price. Needed to detect price changes (Awattar, ..)
+	started                        time.Time    // Boot time
+	updated                        time.Time    // Time of last charged value update
+	gridCharged                    float64      // Grid energy charged since startup (kWh)
+	gridCost                       float64      // Running total of charged grid energy cost (e.g. EUR)
+	gridSavedCost                  float64      // Running total of saved cost from self consumption (e.g. EUR)
+	selfConsumptionCharged         float64      // Self-produced energy charged since startup (kWh)
+	selfConsumptionCost            float64      // Running total of charged self-produced energy cost (e.g. EUR)
+	lastGridPrice, lastFeedInPrice float64      // Stores the last published grid price. Needed to detect price changes (Awattar, ..)
+	persistTicker                  *time.Ticker // Ticker that defines the interval when savings are written to disk
 }
 
 func NewSavings(tariffs tariff.Tariffs) *Savings {
 	clock := clock.New()
 	savings := &Savings{
-		clock:   clock,
-		tariffs: tariffs,
-		started: clock.Now(),
-		updated: clock.Now(),
+		clock:         clock,
+		tariffs:       tariffs,
+		started:       clock.Now(),
+		updated:       clock.Now(),
+		persistTicker: time.NewTicker(time.Minute),
 	}
 
+	savings.restore()
+
 	return savings
 }
 
+func (s *Savings) restore() {
+	load("started", &s.started)
+	load("gridCharged", &s.gridCharged)
+	load("gridCost", &s.gridCost)
+	load("gridSavedCost", &s.gridSavedCost)
+	load("selfConsumptionCharged", &s.selfConsumptionCharged)
+	load("selfConsumptionCost", &s.selfConsumptionCost)
+
+	// fmt.Println("restored savings.*")
+}
+
+func (s *Savings) save() {
+	save("started", s.started)
+	save("gridCharged", s.gridCharged)
+	save("gridCost", s.gridCost)
+	save("gridSavedCost", s.gridSavedCost)
+	save("selfConsumptionCharged", s.selfConsumptionCharged)
+	save("selfConsumptionCost", s.selfConsumptionCost)
+
+	if s.persistTicker == nil {
```

Comment:
Mhhhm- da lass uns nochmal überlegen, warum wir das brauchen. Setting load/store sollte ja egtl. einfacher sein.

Reply:
Bzw.: das kann egtl. in Summe weg. Settings werden ja beim Beenden gespeichert?

Reply:
Hm, nur beim kontrollierten Shutdown halte ich für zu riskant. Bei Stromausfall oder Stecker vom RPI ziehen sind die Daten weg.